src/ToricVarieties/ToricMorphisms/attributes.jl

```diff
@@ -0,0 +1,199 @@
+```
+"""
+image(tm::ToricMorphism) = tm.image
+export image
+
+
+@doc Markdown.doc"""
+    grid_morphism(tm::ToricMorphism)
+
+Return the underlying grid morphism of the toric morphism `tm`.
+
+# Examples
+```jldoctest
+julia> F4 = hirzebruch_surface(4)
+A normal, non-affine, smooth, projective, gorenstein, non-fano, 2-dimensional toric variety without torusfactor
+
+julia> grid_morphism(ToricIdentityMorphism(F4))
+Map with following data
+Domain:
+=======
+Abelian group with structure: Z^2
+Codomain:
+=========
+Abelian group with structure: Z^2
+```
+"""
+grid_morphism(tm::ToricMorphism) = tm.grid_morphism
+export grid_morphism
+
+
+
+@doc Markdown.doc"""
+    morphism_on_torusinvariant_weil_divisor_group(tm::ToricMorphism)
+
+For a given toric morphism `tm`, this method computes the corresponding
+map of the torusinvariant Weil divisors.
+
+# Examples
+```jldoctest
+julia> F4 = hirzebruch_surface(4)
+A normal, non-affine, smooth, projective, gorenstein, non-fano, 2-dimensional toric variety without torusfactor
+
+julia> morphism_on_torusinvariant_weil_divisor_group(ToricIdentityMorphism(F4))
+Map with following data
+Domain:
+=======
+Abelian group with structure: Z^4
+Codomain:
+=========
+Abelian group with structure: Z^4
+```
+"""
+@attr GrpAbFinGenMap function morphism_on_torusinvariant_weil_divisor_group(tm::ToricMorphism)
+
+    # find domain and codomain variety
+    domain_variety = domain(tm)
+    codomain_variety = codomain(tm)
+
+    # find rays of these varieties
+    domain_variety_rays = matrix(ZZ,rays(domain_variety))
+    codomain_variety_rays = matrix(ZZ,rays(codomain_variety))
+
+    # identify the image rays
+    images_of_rays = domain_variety_rays * matrix(grid_morphism(tm))
+
+    # for each image ray, find a max cone in the codomain toric variety which contains it
+    codomain_variety_max_cones = maximal_cones(codomain_variety)
+    rayimage_maxcones_incidence_list = [0 for i in 1:nrows(images_of_rays)]
+    for i in 1:nrows(images_of_rays)
+      for j in 1:length(codomain_variety_max_cones)
+        if contains(codomain_variety_max_cones[j], [images_of_rays[i,k] for k in 1:ncols(images_of_rays)])
+          rayimage_maxcones_incidence_list[i] = j
+          break
+        end
+      end
+    end
+
+    # form list of rays which are in maximal cones of the codomain variety
+    codomain_variety_maxcone_ray_incidence = ray_indices(maximal_cones(codomain_variety))
+    r = rank(character_lattice(codomain_variety))
+    codomain_variety_rays_in_cones = []
+    for i in 1:length(codomain_variety_max_cones)
+        matrix = zeros(Int,nrows(codomain_variety_rays),ncols(codomain_variety_rays))
+        for j in 1:nrows(codomain_variety_rays)
+          if codomain_variety_maxcone_ray_incidence[i,j]
+            for k in 1:ncols(codomain_variety_rays)
+              matrix[j,k] = codomain_variety_rays[j,k]
+            end
+          end
+        end
+        push!(codomain_variety_rays_in_cones, matrix)
+    end
```

Comment:
You can actually use `IncidenceMatrix` from Oscar and build an incidence matrix for the image rays in the target fan in hopefully less code along the lines of
```
julia> fc = normal_fan(cube(2));

julia> fh = normal_fan(hexagon);

julia> rh = collect(rays(fh))
6-element Vector{RayVector{fmpq}}:
 [1, 0]
 [0, 1]
 [1, -1]
 [-1, 1]
 [-1, 0]
 [0, -1]

julia> mc = collect(maximal_cones(fc))
4-element Vector{Cone{fmpq}}:
 A polyhedral cone in ambient dimension 2
 A polyhedral cone in ambient dimension 2
 A polyhedral cone in ambient dimension 2
 A polyhedral cone in ambient dimension 2

julia> image_incidences = IncidenceMatrix(nrays(fh), n_maximal_cones(fc))
6×4 IncidenceMatrix
[]
[]
[]
[]
[]
[]


julia> for i in 1:nrays(fh), j in 1:n_maximal_cones(fc)
       image_incidences[i,j] = contains(mc[j], rh[i])
       end

julia> image_incidences
6×4 IncidenceMatrix
[1, 3]
[1, 2]
[3]
[2]
[2, 4]
[3, 4]

```
Does this maybe simplify this method?